designer/client/components/Visualisation/Lines.tsx

```diff
@@ -16,51 +17,59 @@ type State = {
 };
 
 export class Lines extends React.Component<Props, State> {
+  static contextType = DataContext;
+
   state = {
     showEditor: false,
   };
 
+  constructor(props, context) {
+    super(props, context);
+  }
+
   editLink = (edge: Edge) => {
     this.setState({
       showEditor: edge,
     });
   };
 
   render() {
-    const { layout, data, persona } = this.props;
+    const { layout, persona } = this.props;
+    const { data } = this.context;
 
     return (
       <div>
         <svg height={layout.height} width={layout.width}>
           {layout.edges.map((edge) => {
-            const points = edge.points
-              .map((points) => `${points.x},${points.y}`)
-              .join(" ");
+            const { source, target, points, label } = edge;
+            const pointsString = points.map((p) => `${p.x},${p.y}`).join(" ");
```

Comment:
are you guaranteed points will never be null/undefined?

Reply:
No, but these are values provided by a library. This is unchanged from the code that was previously there, I've just cleaned it up a bit with a destructure